app/javascript/driving/components/ActiveRide.js

```diff
@@ -1,73 +1,124 @@
-                            {ride.from_address}<br />
-                            {ride.from_city}, {ride.from_state} {ride.from_zip}
-                        </p>
-                        <ContactVoter voter_phone_number={ride.voter_phone_number} />
-                        <div className="secondary-info">
-                            <p>Total Passengers: {passengers}</p>
-                            <p>Special requests: {ride.special_requests}</p>
-                        </div>
-                        <div className="bottom-controls secondary">
-                            <button className="btn btn-success btn-api" onClick={()=>this.props.pickupRider(ride)}>Rider picked up</button>
-                            <button className="btn btn-danger btn-api" onClick={()=>this.props.cancelRide(ride)}>Cancel Ride</button>
-                        </div>
-                    </div>
-                )
-            case 'picked_up':
-                mapLink = `https://maps.apple.com/?daddr=${ride.to_address}, ${ride.to_city}, ${ride.to_state}`;
-                return (
-                    <div className="panel panel-full p-y-sm">
-                        <a className="directionsLink" target="_blank" href={mapLink}><i className="fa fa-map-marker"></i> Directions</a>
-                        <label>Dropping off:</label>
-                        <h3>{ride.name}</h3>
-                        <p>
-                            {ride.to_address}<br />
-                            {ride.to_city}, {ride.to_state} {ride.to_zip}
-                        </p>
-                        <ContactVoter voter_phone_number={ride.voter_phone_number} />
-                        <div className="secondary-info">
-                            <p>Total Passengers: {passengers}</p>
-                            <p>Special requests: {ride.special_requests}</p>
-                        </div>
-                        <div className="bottom-controls secondary">
-                            <button className="btn btn-success btn-api" onClick={()=>this.props.completeRide(ride)}>Complete Ride</button>
-                            <button className="btn btn-danger btn-api" onClick={()=>this.props.cancelRide(ride)}>Cancel Ride</button>
-                        </div>
-                    </div>
-                )
-            default:
-                return (
-                    <div className="panel panel-full p-y-sm dispatcher-match">
-                       <h2>Ride with an unknown Status!</h2>
-                    </div>
-                )
-        }
-    }
-}
+const ActiveRide = props => {
+  const { ride, completeRide, pickupRider, cancelRide } = props;
+  let mapLink;
+  const passengers = 1 + parseInt(ride.additional_passengers, 10);
 
+  switch (ride.status) {
+    case 'waiting_acceptance':
+      return <DispatchMatch ride={ride} {...props} />;
+    case 'driver_assigned':
+      mapLink = `https://maps.apple.com/?daddr=${ride.from_address}, ${
```

Comment:
I know this isn't new in this PR, but what happens on Android?

Reply:
On Android these will open in Google Maps.